sdk/src/main/scala/com/horizen/SidechainApp.scala

```diff
@@ -333,24 +333,29 @@ class SidechainApp @Inject()
 
   // Retrieve information for using a web socket connector
   val communicationClient: WebSocketCommunicationClient = new WebSocketCommunicationClient()
-  val webSocketReconnectionHandler: WebSocketReconnectionHandler = new DefaultWebSocketReconnectionHandler(sidechainSettings.websocket)
-
-  // Create the web socket connector and configure it
-  val webSocketConnector : WebSocketConnector with WebSocketChannel = new WebSocketConnectorImpl(
-    sidechainSettings.websocket.address,
-    sidechainSettings.websocket.connectionTimeout,
-    communicationClient,
-    webSocketReconnectionHandler
-  )
+  val webSocketReconnectionHandler: WebSocketReconnectionHandler = new DefaultWebSocketReconnectionHandler(sidechainSettings.websocketClient)
+
+  if(sidechainSettings.websocketClient.enabled) {
+    // Create the web socket connector and configure it
+    val webSocketConnector : WebSocketConnector with WebSocketChannel = new WebSocketConnectorImpl(
+      sidechainSettings.websocketClient.address,
+      sidechainSettings.websocketClient.connectionTimeout,
+      communicationClient,
+      webSocketReconnectionHandler
+    )
 
-  // Start the web socket connector
-  val connectorStarted : Try[Unit] = webSocketConnector.start()
+    // Start the web socket connector
+    val connectorStarted : Try[Unit] = webSocketConnector.start()
+    if (connectorStarted.isSuccess)
+      communicationClient.setWebSocketChannel(webSocketConnector)
+    else if (sidechainSettings.withdrawalEpochCertificateSettings.submitterIsEnabled)
+      throw new RuntimeException("Unable to connect to websocket. Certificate submitter needs connection to Mainchain.")
+
+  } else {
+    log.info("Due to the settings, node is not enabled for connections.")
```

Comment:
Honestly, it is not very clear for which exactly connections. We have a multiple different one.
Let's say something like "Websocket client is disabled."